### DISCUSSION TITLE
Aidex values not broadcasting via API to watchdrip.

### DISCUSSION THREAD

**Discussion by jiresongit** [NightscoutFoundation/xDrip#3035] [Bug reports]
I am trying to use watchdrip with xdrip and GlucoRX Aidex sensor.

It seems that the watchdrip app is not able to receive push notifications from xdrip of BG changes, only when watchdrip itself makes forced requests to xdrip.

In woking with @bigdigital (the author of watchdrip) , it seems that the xdrip Aidex code on the BgReading.bgReadingInsertFromInt has a last parameter of false, meaning that the notification to watchdrip etc does not occur.

I am hoping that I have explained this ok and included as much information as needed.

**Comment by jiresongit** [comment]
Previous discussions on aidex integration https://github.com/NightscoutFoundation/xDrip/discussions/2469 @andyrozman

**Comment by Navid200** [comment]
bigdigital is experienced with xDrip and has created the WatchDrip service in xDrip.
If xDrip needs to be fixed in order for WatchDrip to work, he is the one who needs to make a submission to xDrip.
It defeats the purpose if we make a change that impacts WatchDrip without his direct involvement.

**Reply by Navid200** [reply]
Please use the WatchDrip support channels and submit your bug report there.

**Comment by bigdigital** [comment]
This problem is not directly related to watchdrip API or service. The issue is mostly with the AidexReceiver . When the receiver receives the data via broadcast (from Aidex app as i understand ) the received data is added to the xdrip by using the following function BgReading.bgReadingInsertFromInt(bgValueMgDl, timeStamp, segmentation_timeslice, false); 
Here parameter 4 is defined as **false**. When this parameter  is false there is no call of NewDataObserver.newBgReading(bgReading, is_follower); function . Without this, all external devices and services (pebble, bluejay, broadcast service, etc ) would not be notified about receiving a new data .

**Reply by Navid200** [reply]
Aha!  Thanks

**Reply by Navid200** [reply]
@jiresongit 
Please click on "Create issue from discussion on the right".

**Reply by jiresongit** [reply]
Done, thx
